polkaj-scale/src/main/java/io/emeraldpay/polkaj/scale/reader/UInt128Reader.java

```diff
@@ -5,23 +5,14 @@
 
 import java.math.BigInteger;
 
-public class UInt128Reader implements ScaleReader<BigInteger> {
+public class UInt128Reader extends BigIntReader implements ScaleReader<BigInteger> {
 
-    public static final int SIZE_BYTES = 16;
-
-    public static void reverse(byte[] value) {
-        for (int i = 0; i < value.length / 2; i++) {
-            int other = value.length - i - 1;
-            byte tmp = value[other];
-            value[other] = value[i];
-            value[i] = tmp;
-        }
-    }
+    BigIntReader bir = new BigIntReader();
 
     @Override
     public BigInteger read(ScaleCodecReader rdr) {
```

Comment:
You should refactor this to use constructor injection of the number of bytes a particular read needs and the subclasses can pass in the number of bytes that pertain to them